npeps/npep-173-service-account-matches.md

```diff
@@ -0,0 +1,99 @@
+# NPEP-173: NPEP template
+
+* Issue: [#173](https://github.com/kubernetes-sigs/network-policy-api/issues/173)
+* Status: Provisional
+
+## TLDR
+
+Ability to create policies that control network traffic based on workload identities a.ka.
+[Kubernetes service accounts](https://kubernetes.io/docs/concepts/security/service-accounts/).
+
+## Goals
+
+* Use service accounts (identities) as a way of describing (admin) network policies for pods
+
+## Non-Goals
+
+* There might be other identity management constructs like SPIFFE IDs which
+are outside the scope of this enhancement. We can only provide a way to select
+constructs known to core Kubernetes.
+
+## Introduction
+
+Every pod in Kubernetes will have an associated service account. If user does
+not explicitly provide a service account, a default service account is created
+in the namespace. A given pod will always have 1 service account associated
```

Comment:
Thie things that make SA better than labels:

* Other than mirror pods, every pod has exactly 1 SA
* It can't change over time (simpler management)
* It is bounded in size (as opposed to selectors which can have a large number of terms) which makes it friendlier to on-the-wire implementations
* It's aligned to mesh ideas of identity
* It's often associated with the scope of a Service anyway

It is worse in 1 main way: if I want to quarantine a pod, how?  Today I can just change a label.